lxd-migrate/main_netcat.go

```diff
@@ -51,22 +50,26 @@ func (c *cmdNetcat) Run(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	// We'll wait until we're done reading from the socket
-	wg := sync.WaitGroup{}
-	wg.Add(1)
-
+	errs := make(chan error, 2)
 	go func() {
-		io.Copy(eagain.Writer{Writer: os.Stdout}, eagain.Reader{Reader: conn})
-		conn.Close()
-		wg.Done()
-	}()
+		_, err := io.Copy(eagain.Writer{Writer: os.Stdout}, eagain.Reader{Reader: conn})
+		errs <- err
 
-	go func() {
-		io.Copy(eagain.Writer{Writer: conn}, eagain.Reader{Reader: os.Stdin})
+		errs <- conn.Close()
+
+		defer close(errs)
 	}()
 
-	// Wait
-	wg.Wait()
+	_, err = io.Copy(eagain.Writer{Writer: conn}, eagain.Reader{Reader: os.Stdin})
```

Comment:
@markylaing ok would be good to test lxd-migrate works with this as there are no tests for it sadly.

Reply:
@tomponline I've tested lxd-migrate manually and found that moving the second copy out of it's goroutine causes it to block the main thread. I think that's because it is copying from os.Stdin so there is no EOF. I've now returned it to it's own goroutine and restored the wait group. The only error that is now being checked is the one from the first copy operation, for which I'm just setting the value of err (it's value will be populated after wg.Wait() is called).

Reply:
OK thanks, that'll be fine for now, the approach I used would have required the stdin to be closed on error of the other io.Copy.